pkg/kubelet/status/status_manager.go

```diff
@@ -1038,8 +1038,18 @@ func normalizeStatus(pod *v1.Pod, status *v1.PodStatus) *v1.PodStatus {
 		normalizeContainerState(&cstatus.State)
 		normalizeContainerState(&cstatus.LastTerminationState)
 	}
-	// Sort the container statuses, so that the order won't affect the result of comparison
+	// Sort the init container statuses, so that the order won't affect the result of comparison
 	kubetypes.SortInitContainerStatuses(pod, status.InitContainerStatuses)
+
+	// update ephtemeral container statuses
+	for i := range status.EphemeralContainerStatuses {
```

Comment:
I think we can use one function to handle `InitContainerStatuses` `EphemeralContainerStatuses` and `ContainerStatuses` so that we can avoid duplication of code.

Reply:
Agreed, thx~